### PR TITLE
introduce epoch

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

**Comment by ngaut** [comment]
PTAL @siddontang @BusyJay

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -25,6 +25,7 @@ message RaftMessage {
     optional metapb.Peer from_peer    = 2;
     optional metapb.Peer to_peer      = 3;
     optional raftpb.Message message   = 4;
+    optional metapb.RegionEpoch region_epoch = 5;
```
Comment (BusyJay) [inline proto/raft_serverpb.proto]: This number in this line is not aligned with the lines above.

**Comment by BusyJay** [comment]
rest LGTM

File: proto/raft_serverpb.proto (hunk 1 above)
Comment (siddontang) [inline proto/raft_serverpb.proto]: +1

**Comment by siddontang** [comment]
Rest LGTM

**Comment by ngaut** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**